src/components/MyAccount/RequestsTab/UpdateLocation.tsx

```diff
@@ -0,0 +1,196 @@
+          method: "PUT",
+          body: JSON.stringify({
+            pickupLocation: selected.code,
+            patronId: `${patronId}`,
+          }),
+        }
+      )
+      if (response.status == 200) {
+        // Open next modal to confirm request has been canceled.
+        setModalProps(successModalProps(selected) as DefaultModalProps)
+      } else setModalProps(failureModalProps as DefaultModalProps)
+    },
+    headingText: (
+      <Heading className={styles.modalHeading}>
+        <Text sx={{ marginBottom: 0 }}>
+          {" "}
+          Where would you like to pick up this item?{" "}
+        </Text>
+      </Heading>
+    ),
+  })
+
+  const [modalProps, setModalProps] = useState<BaseModalProps>(
+    confirmLocationChangeModalProps(pickupLocation) as ConfirmationModalProps
+  )
+
+  const successModalProps = (newLocation: SierraCodeName) => ({
+    type: "default",
+    bodyContent: (
+      <Box className={styles.modalBody}>
+        <Text sx={{ marginLeft: "l" }}>
+          Your item will be available for pickup at the {newLocation.name}{" "}
+          Library.
+        </Text>
+      </Box>
+    ),
+    closeButtonLabel: "OK",
+    headingText: (
+      <Heading className={styles.modalHeading}>
+        <>
+          <Icon
+            size="large"
+            name="actionCheckCircleFilled"
+            color="ui.success.primary"
+          />
+          <Text sx={{ marginBottom: 0 }}> Location change successful </Text>
+        </>
+      </Heading>
+    ),
+    onClose: () => {
+      updateHoldLocation(holdId, newLocation)
+      setModalProps(
+        confirmLocationChangeModalProps(newLocation) as ConfirmationModalProps
+      )
+      closeModal()
+    },
+  })
+  const failureModalProps = {
+    type: "default",
+    bodyContent: (
+      <Box className={styles.modalBody}>
+        <Text sx={{ marginLeft: "l", marginRight: "m" }}>
+          We were unable to change the pickup location. Please contact{" "}
+          <DSLink href="https://www.nypl.org/get-help/contact-us">
+            AskNYPL
+          </DSLink>{" "}
+          for further assistance.
+        </Text>
+      </Box>
+    ),
+    onClose: () => {
+      setModalProps(
+        confirmLocationChangeModalProps(
+          pickupLocation
+        ) as ConfirmationModalProps
+      )
+      closeModal()
+    },
+    closeButtonLabel: "OK",
+    headingText: (
+      <Heading className={styles.modalHeading}>
+        <>
+          <Icon size="large" name="errorFilled" color="ui.error.primary" />
+          <Text sx={{ marginBottom: 0 }}> Location change failed </Text>
+        </>
+      </Heading>
+    ),
+  }
+  return (
+    <>
+      <Button
+        sx={{ paddingLeft: 0 }}
+        size="small"
+        onClick={openModal}
+        id={`update-pickup-location-${key}`}
+        buttonType="text"
+      >
+        <Icon name="editorMode" align="left" size="small"></Icon>
+        <Text className={styles.changeLocation}>Change location</Text>
```

Comment:
nit but I think the text of this button needs to be bigger

Reply:
update the size